src/library/scala/collection/convert/Accumulator.scala

```diff
@@ -37,7 +37,9 @@ trait Accumulator[@specialized(Double, Int, Long) A, +CC[_], +C]
     else 1 << 24
   }
 
-  final override def size: Int = if (longSize < Int.MaxValue) longSize.toInt else Int.MaxValue /* not sure.. default implementation just overflows */
+  final override def size: Int =
+    if (longSize < Int.MaxValue) longSize.toInt
+    else throw new IllegalArgumentException(s"Size too large for an Int: $longSize")
```

Comment:
Whoa, you can't throw an exception here unless you fix `knownSize` to not defer to `size` any longer!  The whole point of `knownSize` is to be safe when the size isn't known/knowable.

Reply:
Ah, very good point, thank you. It just feels to me that returning an incorrect `size` is worse than throwing. So I change `knownSize` to return `-1` in case the `longSize` is too long.